configure-ironic.sh

```diff
@@ -2,6 +2,14 @@
 
 . /bin/ironic-common.sh
 
+USE_HTTP_BASIC=${USE_HTTP_BASIC:-false}
```

Comment:
We can use non-empty username/password as a flag.

Reply:
if everyone is ok with using ``IRONIC_HTTP_BASIC_USERNAME`` and  ``IRONIC_HTTP_BASIC_PASSWORD`` to say that `http_basic` will be used instead of `noauth` I will update, let's get more feedback for this one

Reply:
Do we need an explicit mode variable to control which of the 3 planned approaches (noauth, basic, and mTLS) is used?

Reply:
one variable wouldn't handle I think

Reply:
I'd still prefer to avoid redundant variables. The current approach allows for USE_HTTP_BASIC=true to end up with insecure defaults.

Reply:
@dtantsur I will push a new PR, wdyt about `IRONIC_AUTH_STRATEGY` with values (`noauth`, `http_basic`) and the user/password default's to empty string and we fail if `http_basic` is set and either user/password is empty?

Reply:
Works for me